File: rules/pascal/tests/pas.cmd-injection/cmd.pas
program CmdTests;
var s: String;
begin
  s := ParamStr(1);
  WinExec(PChar('cmd /c ' + s), 0); // crit:expect pas.cmd-injection
  ShellExecute(0, 'open', PChar(ParamStr(2)), nil, nil, 0); // crit:expect pas.cmd-injection
  WinExec(PChar('cmd /c dir'), 0); // crit:expect-not pas.cmd-injection
  ShellExecute(0, 'open', 'notepad.exe', nil, nil, 0); // crit:expect-not pas.cmd-injection
end.

File: rules/csharp/tests/cs.cmd-injection/Cmd.cs
class CmdTests {
  void Direct() {
    var id = Request.QueryString["id"];
    Process.Start("cmd", "/c " + id); // crit:expect cs.cmd-injection
  }
  void ViaForm() {
    var host = Request.Form["host"];
    Process.Start("ping " + host); // crit:expect cs.cmd-injection
  }
  void Fixed() {
    Process.Start("notepad.exe"); // crit:expect-not cs.cmd-injection
  }
  void Internal(string arg) {
    Process.Start("cmd", arg); // crit:expect-not cs.cmd-injection
  }
}
